core/includes/install.inc

```diff
@@ -454,31 +456,29 @@ abstract class DatabaseTasks {
     //   socket may work instead. This can happen when using local environments
     //   that use non-standard port numbers for the database.
     if ($last_exception && ($last_exception->getCode() === $this->connectionRefusedErrorCode)) {
-      if ($original_connection_info['default']['host'] === 'localhost') {
+      $localhost_invalid = $original_connection_info['default']['host'] === 'localhost';
+      $local_ip_invalid = $original_connection_info['default']['host'] === '127.0.0.1';
+      if ($localhost_invalid) {
         $modified_connection_info['default']['host'] = '127.0.0.1';
         $modified_connection_info['default']['port'] = '3306';
       }
-      elseif ($original_connection_info['default']['host'] === '127.0.0.1') {
+      elseif ($local_ip_invalid) {
         $modified_connection_info['default']['host'] = 'localhost';
         $modified_connection_info['default']['port'] = '';
       }
       if ($original_connection_info != $modified_connection_info) {
         Database::removeConnection('default');
         Database::addConnectionInfo('default', 'default', $modified_connection_info['default']);
         try {
-          Database::getConnection();
-          $last_exception = FALSE;
-
-          // Connection was successful, but settings.php needs to be updated.
-          $db_settings = $modified_connection_info['default'];
-          $settings['database'] = array(
-            'value' => $db_settings['driver'] . '://' . rawurlencode($db_settings['username']) . ':' . rawurlencode($db_settings['password']) . '@' . $db_settings['host'] . ($db_settings['port'] ? (':' . $db_settings['port']) : '') . '/' . rawurlencode($db_settings['database']),
-            'required' => TRUE,
-          );
-          $settings['database_prefix'] = array(
-            'value' => $db_settings['prefix']['default'],
-          );
-          backdrop_rewrite_settings($settings);
```

Comment:
I modified this section because `backdrop_rewrite_settings()` was getting called twice on the `$database` settings, causing the value to become invalid PHP. Instead of rewriting automatically in this unexpected place, now a better message is shown instead.

Reply:
Makes sense. Thanks